com.unity.sg2/Editor/GraphDeltaRegistry/Registry/ShaderGraphRegistry.cs

```diff
@@ -117,34 +135,45 @@ internal void InitializeDefaults()
                 {
                     var ndMethod = t.GetMethod(GET_ND_METHOD_NAME);
                     var fdMethod = t.GetMethod(GET_FD_METHOD_NAME);
-                    var udMethod = t.GetMethod(GET_UD_METHOD_NAME);
+                    var uidMethod = t.GetMethod(GET_UD_METHOD_NAME);
+                    var versionMethod = t.GetMethod(GET_VERSION_METHOD_NAME);
 
+                    // has NodeDescriptor
                     if (ndMethod != null)
                     {
                         var nd = (NodeDescriptor)ndMethod.Invoke(null, null);
                         if (!nd.Equals(default(NodeDescriptor)))
-                        { // use the NodeDescriptor
-                            if (udMethod != null)
+                        {
+                            if (uidMethod != null)
                             {
-                                var ui = (NodeUIDescriptor)udMethod.Invoke(null, null);
+                                var ui = (NodeUIDescriptor)uidMethod.Invoke(null, null);
                                 Register(nd, ui);
                             }
-                            else Register(nd);
+                            else
+                                Register(nd);
                         }
                     }
+                    // has FunctionDescriptor
                     else if (fdMethod != null)
                     {
+                        int version = 8;
+                        if (versionMethod != null)
+                        {
+                            version = (int)versionMethod.Invoke(null, null);
+                        }
                         var fd = (FunctionDescriptor)fdMethod.Invoke(null, null);
                         if (!fd.Equals(default(NodeDescriptor)))
-                        {  // use the FunctionDescriptor
-                            if (udMethod != null)
+                        {
+                            if (uidMethod != null)
```

Comment:
I'm not even sure if this is a valid nit; the `uidMethod` case here and on line 147 can probably be pulled out to the root level if you wanted, just so the `uidMethod.Invoke` is only referenced once in this block of conditions.

Reply:
I noticed this. I didn't change it, only its name.

Reply:
Actually, I think that the processing ends up being the same because there is still a null test after the Invoke. It only gets invoked once.